app/screens/scheduled_post_options/scheduled_post_picker.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {Platform, View} from 'react-native';
+
+import FormattedText from '@components/formatted_text';
+import {Screens} from '@constants';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import BottomSheet from '@screens/bottom_sheet';
+import ScheduledPostCoreOptions from '@screens/scheduled_post_options/core_options';
+import ScheduledPostFooter from '@screens/scheduled_post_options/footer';
+import {logInfo} from '@utils/log';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {getTimezone} from '@utils/user';
+
+import type {BottomSheetFooterProps} from '@gorhom/bottom-sheet';
+import type UserModel from '@typings/database/models/servers/user';
+
+const OPTIONS_PADDING = 12;
+const OPTIONS_SEPARATOR_HEIGHT = 1;
+const TITLE_HEIGHT = 54;
+const ITEM_HEIGHT = 48;
+
+export const SCHEDULED_POST_OPTIONS_BUTTON = 'close-scheduled-post-options';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        backgroundColor: theme.centerChannelBg,
+        height: 200,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        ...typography('Heading', 600, 'SemiBold'),
+    },
+    optionsContainer: {
+        paddingTop: OPTIONS_PADDING,
+    },
+    optionsSeparator: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: OPTIONS_SEPARATOR_HEIGHT,
+    },
+}));
+
+type Props = {
+    currentUser?: UserModel;
+    serverUrl: string;
+}
+
+export function ScheduledPostOptions({currentUser}: Props) {
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const [selectedTime, setSelectedTime] = useState<string | null>(null);
+
+    const userTimezone = getTimezone(currentUser?.timezone);
+
+    const style = getStyleSheet(theme);
+
+    const snapPoints = useMemo(() => {
+        const bottomSheetAdjust = Platform.select({ios: 5, default: 20});
+        const numberOfItems = 11;
```

Comment:
We don't have 11 items. We have a variable number of options, and the time picker, right? We may want to be a bit more specific about this.

Reply:
Fixed, added comment explaining the same.